src/ORM/RulesChecker.php

```diff
@@ -113,6 +114,47 @@ public function existsIn($field, $table, $message = null)
     }
 
     /**
+     * Returns a callable that can be used as a rule for checking that the entity is not
+     * referenced by any row in the specified table
+     *
+     * This is useful for enforcing foreign key integrity checks. (i.e.: ON DELETE RESTRICT)
+     *
+     * ### Example:
+     *
+     * ```
+     * $rules->add($rules->isNotReferencedBy('Articles'));
+     * ```
+     *
+     * @param null|string|array $field The field or list of fields that are the foreign keys in $table
+     * linking to the this table's primary key
+     * @param object|string $table The table name to check for references
+     * @param string|array|null $message The error message to show in case the rule does not pass. Can
+     *   also be an array of options. When an array, the 'message' key can be used to provide a message.
+     * @return callable
+     */
+    public function isNotReferencedBy($field, $table, $message = null)
+    {
+        $options = [];
+        if (is_array($message)) {
+            $options = $message + ['message' => null];
+            $message = $options['message'];
+            unset($options['message']);
+        }
+
+        if (!$message) {
+            if ($this->_useI18n) {
+                $message = __d('cake', 'This field is referenced by another entity');
+            } else {
+                $message = 'This field is referenced by another entity';
+            }
+        }
+
+        $errorField = is_string($field) ? $field : current($field);
+
+        return $this->_addError(new IsNotReferencedBy($field, $table, $options), '_isNotReferencedBy', compact('errorField', 'message'));
```

Comment:
@markstory yes I'm not finished with this PR yet, just didn't have time to finish up yet.